ProcessLib/BoundaryCondition/HCNonAdvectiveFreeComponentFlowBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,121 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "GenericNaturalBoundaryConditionLocalAssembler.h"
+#include "MeshLib/Elements/MapBulkElementPoint.h"
+#include "MeshLib/PropertyVector.h"
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "NumLib/Function/Interpolation.h"
+#include "ParameterLib/MeshNodeParameter.h"
+#include "ProcessLib/Process.h"
+
+namespace ProcessLib
+{
+struct HCNonAdvectiveFreeComponentFlowBoundaryConditionData
+{
+    ParameterLib::Parameter<double> const& boundary_permeability;
+    std::vector<std::size_t> const bulk_face_ids;
+    std::vector<std::size_t> const bulk_element_ids;
+    Process const& process;
+};
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+class HCNonAdvectiveFreeComponentFlowBoundaryConditionLocalAssembler final
+    : public GenericNaturalBoundaryConditionLocalAssembler<
+          ShapeFunction, IntegrationMethod, GlobalDim>
+{
+    using Base = GenericNaturalBoundaryConditionLocalAssembler<
+        ShapeFunction, IntegrationMethod, GlobalDim>;
+    using NodalVectorType = typename Base::NodalVectorType;
+    using NodalMatrixType = typename Base::NodalMatrixType;
+
+public:
+    /// The neumann_bc_term factor is directly integrated into the local
+    /// element matrix.
+    HCNonAdvectiveFreeComponentFlowBoundaryConditionLocalAssembler(
+        MeshLib::Element const& e,
+        std::size_t const local_matrix_size,
+        bool const is_axially_symmetric,
+        unsigned const integration_order,
+        HCNonAdvectiveFreeComponentFlowBoundaryConditionData const& data)
+        : Base(e, is_axially_symmetric, integration_order),
+          _data(data),
+          _local_matrix_size(local_matrix_size),
+          _bulk_element_id(data.bulk_element_ids[e.getID()]),
+          _bulk_face_id(data.bulk_face_ids[e.getID()]),
+          _surface_eigenvector(giveEigenvector(e))
+    {
+    }
+    Eigen::RowVector3d giveEigenvector(MeshLib::Element const& e)
```

Comment:
The name of the method is rather generic. Is my understanding correct that the method returns the surface normal? Then the name could also reflect the nature of the returned vector. Maybe `getSurfaceNormal()`?

Reply:
Changed